ACE/ace/Malloc_T.cpp

```diff
@@ -190,9 +190,16 @@ ACE_Cascaded_Dynamic_Cached_Allocator<ACE_LOCK>::ACE_Cascaded_Dynamic_Cached_All
   // If ACE_NEW fails, the hierarchy_ will be reconstructed when malloc API is called.
   ACE_NEW (tmp, comb_alloc_type(this->initial_n_chunks_, this->chunk_size_));
 
-  // Increase the chunk sum if succeed.
-  this->chunk_sum_ += tmp->pool_depth();
+  // Consider the exception of vector push_back call
   this->hierarchy_.push_back(tmp);
+  if (0 == this->hierarchy_.size())
```

Comment:
when push_back fails it will throw, maybe use std::unique_ptr?

Reply:
By the way,  I can see the result of these calls to judge whether potential exceptions have occurred

Reply:
See https://en.cppreference.com/w/cpp/language/exceptions, push_back can throw, when it does the state of the std::vector is guaranteed at that moment, but it will not just return. The exception should go back to the caller to my idea, your allocator should not leak when push_back (or any operation it uses) throws

Reply:
when using unique_ptr the check after the push_back seems invalid

Reply:
I have used std::unique_ptr for idiom in new commit

But for a vector of pointer, I think sdt::vector can keep strong exception guarantee.

> If an exception is thrown (which can be due to Allocator::allocate() or element copy/move constructor/assignment), this > function has no effect ([strong exception guarantee]> (https://en.cppreference.com/w/cpp/language/exceptions#Exception_safety)).

---

> Strong exception guarantee — If the function throws an exception, the state of the program is rolled back to the state > just before the function call (for example, [std::vector::push_back]>(https://en.cppreference.com/w/cpp/container/vector/push_back)).

Reply:
Sorry, Your idea is right!

The APIs provide strong exception guarantee, but not Nothrow 

I will delete the defence codes